### PR TITLE
✨ Adds support for React.FC & FC

### PR DESCRIPTION
**Pull request by danieldelcore**
Closes: #181 

CC: @mengX666

### CODE REVIEW

File: packages/extract-react-types/src/utils.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable import/prefer-default-export */
```
Comment (pgmanutd) [inline packages/extract-react-types/src/utils.js:1]: This eslint rule creates lot of noise in repos. We can disable it altogether.

File: packages/extract-react-types/src/utils.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+export function hasTypeAnnotation(path, left, right) {
```
Comment (pgmanutd) [inline packages/extract-react-types/src/utils.js:3]: I really like this function ❤️. Got lost in `if` checks though 😅

File: packages/extract-react-types/src/utils.js (hunk 1 above)
Comment (danieldelcore) [reply packages/extract-react-types/src/utils.js:1]: Yeah i think we should kill this one too!

File: packages/extract-react-types/src/utils.js (hunk 2 above)
Comment (danieldelcore) [reply packages/extract-react-types/src/utils.js:3]: Yeah it's a bit of a hairy one, isn't it 🤔 
Can you see any obvious ways to simplify it? Maybe a change of approach might be best here instead.

Comment (pgmanutd) [reply packages/extract-react-types/src/utils.js:3]: Yeah! Don't have full context but can we do something like this (not sure if this is correct, just a nit TBH 😅):
We can also skip `try` `catch` I think 🤔

```js
const { typeName, typeParameters } = path.get('id.typeAnnotation.typeAnnotation').node;

return !!typeParameters && [
  typeName.left?.name === left && typeName.right?.name === right,
  typeName.name === left || typeName.name === right,
].some(Boolean);
```

Comment (danieldelcore) [reply packages/extract-react-types/src/utils.js:3]: Updated 😄 

I managed to remove the `try/catch` and simplify the logic a bit. 

Hopefully it's looking better. LMK what you think

**Review by pgmanutd — APPROVED** [review]
lgtm 🎉

**Pull request merged**